### PR TITLE
remove --pull and --branch from bootstrap.py, redux

### PR DESCRIPTION
**Pull request by BenTheElder**
fix of #4482. sorta gross but this makes sure the tests actually flex `parse_args` and makes sure `--pull` and `--branch` stay unset.

Also adds myself and @krzyzacy to code owners so we can be notified of any future modifications to bootstrap.py / bootstrap_test.py

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -987,6 +980,10 @@ def parse_args(arguments=None):
         action='store_true',
         help='Clean the git repo before running tests.')
     args = parser.parse_args(arguments)
+    # --pull is deprecated, use --repo=k8s.io/foo=master:abcd,12:ef12,45:ff65
+    setattr(args, 'pull', None)
+    # --branch is deprecated, use --repo=k8s.io/foo=master
+    setattr(args, 'branch', None)
```
Comment (BenTheElder) [inline jenkins/bootstrap.py]: per https://github.com/kubernetes/test-infra/pull/4482#issuecomment-328648313
This is a bit gross, but it should suffice to prevent usage while allowing a lot of otherwise valid but somewhat hairy tests to work until the port is in place.

File: jenkins/bootstrap_test.py [hunk 2]
```diff
@@ -394,45 +394,8 @@ def test_happy(self):
...
-    def test_depre_pull_repo_commit(self):
-        """--repo=foo=master:aaa --pull=123:abc is not allowed"""
-        with self.assertRaises(ValueError):
-            bootstrap.parse_repos(FakeArgs(
-                repo=['foo=master:aaa'], branch='', pull='123:abc'))
-        with self.assertRaises(ValueError):
-            bootstrap.parse_repos(FakeArgs(
-                repo=['foo=master'], branch='', pull='123:abc'))
+        args = bootstrap.parse_args(["--job=foo", "--bare"])
+        self.assertFalse(bootstrap.parse_repos(args))
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: how about add another test with `--branch`/`--pull` and assert they will be set to `None`?

(I still

Comment (BenTheElder) [reply jenkins/bootstrap_test.py]: Done.

File: jenkins/bootstrap_test.py [hunk 3]
```diff
@@ -394,45 +394,14 @@ def test_happy(self):
...
+
+    def test_pull_branch_none(self):
+        """args.pull and args.branch should be None"""
+        args = bootstrap.parse_args(["--job=foo", "--bare"])
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: pass in `--branch` and `--pull`?

File: jenkins/bootstrap_test.py [hunk 4]
```diff
@@ -394,45 +394,14 @@ def test_happy(self):
...
+        self.assertIsNone(args.pull)
+        self.assertIsNone(args.branch)
 
     def test_plain(self):
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: one last nit: I would unify the `parse_repos` call in `ParseReposTest` not using `FakeArgs`, as the rest are all `branch=''` and `pull=''` now. (as `FakeArgs` are evil, and this unit-test is still sorta important to have)

Comment (BenTheElder) [reply jenkins/bootstrap_test.py]: done

**Comment by krzyzacy** [comment]
/lgtm

fix pylint

**Comment by ixdy** [comment]
In addition to the maintenance job (being fixed in #4569) I found a few other places not yet fixed:
https://github.com/kubernetes/test-infra/blob/738c3442bba99088268e9fe95727e413cde6b986/images/pull-test-infra-gubernator/runner#L24
https://github.com/kubernetes/test-infra/blob/738c3442bba99088268e9fe95727e413cde6b986/experiment/jenkins_to_prow.py#L109

**Comment by BenTheElder** [comment]
Noting that these have been fixed now. Hopefully there should be no more `--pull` or `--branch` anywhere in test-infra except `--pull` for some Docker stuff...
